### DISCUSSION TITLE
i18n-js CLI 4.0.0.alpha4 has been released!

### DISCUSSION THREAD

**Discussion by fnando** [fnando/i18n-js#654] [Announcements]
Hey everyone! It's been a while!

I'm going to release the final i18n-js v4 next week. As many of you already know, this is a *breaking release*. So, if you won't migrate to v4, make sure you have your project pinned down to v3. Notice that v3 won't receive any further development efforts.

We have a [new release up on RubyGems](https://rubygems.org/gems/i18n-js/versions/4.0.0.alpha4). What's new, you may ask. Here it is:

### Checking for missing translations

The `i18n check` command will check your loaded translations and output the keys that don't have an equivalent key against the default locale. Here's what it looks like:

![`i18n check` command in action](https://github.com/fnando/i18n-js/raw/main/images/i18njs-check.gif)

### Add digest to exported files' name

Your configuration can now have `:digest` as part of exported files. This will generate a cache-bursting file with the embedded md5 hex digest.

---

That is it! Take care!

Nando V.

**Comment by ryanb** [comment]
@fnando Thanks for your work on this!

Is there an upgrade guide somewhere?

**Reply by fnando** [reply]
Hey Ryan, I didn't had the chance to write this yet, but the gist of the process is:

1. `yarn add i18n-js` to add the JavaScript dependency
2. Update `config/i18n.yml` to use the new pattern for matching files (check the readme file)
3. For development, choose a way to track changes on your files and re-export the translations (I'm using watchman for a bunch of stuff, so that's my personal choice, but you can use guard or listen for that as well; there's a section about this on the readme file).
4. For production, you can either save the exported translations (I don't do this) or export translations using `i18n export` during your build step.

Feel free to ping this thread if you find any issues while upgrading.

**Reply by fnando** [reply]
Just wrote a proper document, but it may still be missing some migration info. If you find any issues during the migration, let me know and I'll update the document. ;)

https://github.com/fnando/i18n-js/blob/main/MIGRATING_FROM_V3_TO_V4.md

**Reply by ryanb** [reply]
Thanks! I plan to try this out near the end of this week.
